pkg/addonfactory/addonfactory.go

```diff
@@ -0,0 +1,35 @@
+package addonfactory
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"open-cluster-management.io/addon-framework/pkg/agent"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+const AddonDefaultInstallNamespace = "open-cluster-management-agent-addon"
+
+// AnnotationValuesName is the annotation Name of customized values
+const AnnotationValuesName string = "addon.open-cluster-management.io/values"
+
+// BuiltinValues includes the built-in values
+// for helm chart, the values should begin with a lowercase letter, so we need convert it to Values by JsonStructToValues.
```

Comment:
I'd rather we do not use the same struct for both helm and template.

Reply:
done. defined different struct for helm and template